src/AaveV2CollectorContractConsolidation.sol

```diff
@@ -0,0 +1,144 @@
+
+    /// Not enough token left to swap
+    error NotEnoughTokens();
+    /// Oracle price is 0 or lower
+    error InvalidOracleAnswer();
+    /// Need to request more than 0 tokens out
+    error OnlyNonZeroAmount();
+    /// Token is not available for swap
+    error UnsupportedToken();
+
+    struct Asset {
+        uint256 quantity;
+        uint48 premium;
+        address oracle;
+        bool ethFeedOnly;
+    }
+
+    mapping(address => Asset) public assets;
+
+    constructor() {
+        assets[ARAI] = Asset(6740239e16, 100, 0x483d36F6a1d063d580c7a24F9A42B346f3a69fbb, false); // Custom Feed
+        assets[AAMPL] = Asset(15891248e7, 300, 0xe20CA8D7546932360e37E9D72c1a47334af57706, false); // Monitored Feed
+        assets[AFRAX] = Asset(2869022e16, 75, 0xB9E1E3A9feFf48998E45Fa90847ed4D467E8BcfD, false);
+        assets[FRAX] = Asset(1512479102e11, 75, 0xB9E1E3A9feFf48998E45Fa90847ed4D467E8BcfD, false);
+        assets[AENS] = Asset(7047e16, 300, 0x5C00128d4d1c2F4f652C267d7bcdD7aC99C16E16, false);
+        assets[SUSD] = Asset(9040e16, 75, 0x8e0b7e6062272B5eF4524250bFFF8e5Bd3497757, true); // Only ETH
+        assets[ASUSD] = Asset(1148320e16, 75, 0x8e0b7e6062272B5eF4524250bFFF8e5Bd3497757, true); // Only ETH
+        assets[TUSD] = Asset(160409e16, 75, 0xec746eCF986E2927Abd291a2A1716c940100f8Ba, false);
+        assets[ATUSD] = Asset(608004e16, 75, 0xec746eCF986E2927Abd291a2A1716c940100f8Ba, false);
+        assets[AMANA] = Asset(1622740e16, 200, 0x56a4857acbcfe3a66965c251628B1c9f1c408C19, false);
+        assets[MANA] = Asset(33110e16, 200, 0x56a4857acbcfe3a66965c251628B1c9f1c408C19, false);
+        assets[ADPI] = Asset(1359e16, 300, 0xD2A593BF7594aCE1faD597adb697b5645d5edDB2, false); 
+        assets[ABUSD] = Asset(364085e16, 75, 0x833D8Eb16D306ed1FbB5D7A2E019e106B960965A, false);
+        assets[BUSD] = Asset(3399064e14, 75, 0x833D8Eb16D306ed1FbB5D7A2E019e106B960965A, false);
+        assets[ZRX] = Asset(10719e16, 300, 0x2885d15b8Af22648b98B122b22FDF4D2a56c6023, false);
+        assets[AZRX] = Asset(877140e16, 300, 0x2885d15b8Af22648b98B122b22FDF4D2a56c6023, false);
+        assets[AUST] = Asset(89239797e4, 200, address(0), false); // NO FEED
+        assets[ARENFIL] = Asset(41067e16, 300, address(0), false); // NO FEED
+    }
+
+    /// @notice Swaps USDC for specified token
+    /// @param _token the address of the token to swap USDC for
+    /// @param _amountOut the amount of token wanted
+    /// @dev User has to approve USDC transfer prior to calling swap 
+    function swap(address _token, uint256 _amountOut) external {
+        if (_amountOut == 0) revert OnlyNonZeroAmount();
+
+        uint256 amountIn = getAmountIn(_token, _amountOut, ERC20(_token).decimals());
+
+        uint256 quantity = assets[_token].quantity;
+        uint256 sendAmount = _amountOut == type(uint256).max ? quantity : _amountOut;
+
+        unchecked {
+            assets[_token].quantity = quantity - sendAmount;
+        }
```

Comment:
How much gas are you saving by doing this?

Reply:
I can add a test case but it should not overflow since the sendAmount is being checked elsewhere against the quantity already.

It saves 70 gas total. Might not be worth it for this contract as we don't expect that many calls, but I feel ok about this one. I can go either way because the number of calls will be few.